fluvii/kafka_tools/fluvii_toolbox.py

```diff
@@ -92,51 +93,107 @@ def create_topics(self, topic_config_dict, ignore_existing_topics=True):
                 futures[topic].result()
         LOGGER.info(f'Created topics: {list(topic_config_dict.keys())}')
 
-    def alter_topics(self, topic_config_dict, retain_configs=True, ignore_missing_topics=True, protected_configs=None):
+    def alter_topics(self, topic_config_dict, retain_configs=True, ignore_missing_topics=True, ignore_field_list=None):
         """
-        {'topic_a': {'partitions': 1, 'replication.factor': 1, 'segment.ms': 10000}, 'topic_b': {etc}}
+        {'topic_a': {'segment.ms': 10000}, 'topic_b': {etc}}
+        NOTE: you cannot change partitions or replication factor this way
+        NOTE: "retain_configs" means keeping a previous value if you did not provide a new value for a given setting.
+        TODO: make a preview option
         """
         self._set_admin()
-        if not protected_configs:
-            protected_configs = []
+        if not ignore_field_list:
+            ignore_field_list = []
+        ignore_field_list += ['partitions', 'replication_factor', 'replication.factor']
+        ignore_field_list = set(ignore_field_list)
+
         current_configs = {}
         if retain_configs:
-            current_configs = self.list_topics(include_configs=True)
-            for topic in current_configs.items():
-                current_configs[topic] = {k: v for k, v in current_configs[topic].items() if k not in protected_configs}
+            current_configs = {t: cfgs for t, cfgs in self.list_topics(include_configs=True).items() if t in topic_config_dict}
             topics = current_configs.keys()
         else:
             topics = self.list_topics()
+
         if ignore_missing_topics:
             existing = set(topics)
             missing = set(topic_config_dict.keys()) - existing
             if missing:
                 LOGGER.info(f'These topics dont exist, ignoring: {missing}')
                 for i in missing:
                     topic_config_dict.pop(i)
+            if not topic_config_dict:
+                LOGGER.info('No valid topics.')
```

Comment:
Eh, I could see you just trying to create topics as a sanity check...dunno if it needs to raise exceptions necessarily. I can see the argument though.

Reply:
It does

Reply:
Would you say that this is error reason why it does?

Reply:
Wanna be sure I follow: are you suggesting that it should raise an exception, or just log it via .error or something?

If the former and you ran a sync command, it could be no topics need to be made, and raising an exception would kill the sync.

Reply:
Yes, I'm talking about adjusting the log level above info. I see now though that all topics could already be synced which would only need to be informative rather than a warning, error, or fatal.

The log message mentions that there were no _valid_ topics. I was imagining a case where the user made a mistake somehow with their topic configuration and glazed over any info logging or had fluvii only configured for warn or above.

Reply:
I see, it's a wording thing. I could just say "No topics to create"

Reply:
I re-worded this :+1: